frontend/rolecall/src/app/api/cast_api.service.ts

```diff
@@ -192,95 +199,73 @@ export class CastApi {
-  async requestCastSet(cast: Cast): Promise<HttpResponse<any>> {
-    if (environment.mockBackend) {
-      return this.mockBackend.requestCastSet(cast);
-    }
-    // Check if we have record of the cast and patch if we do
-    if (this.hasCast(cast.uuid)) {
-      // Do patch
-      let header = await this.headerUtil.generateHeader();
-      return this.http.delete(environment.backendURL + 'api/cast?castid=' + cast.uuid, {
+/** Hits backend with create/edit cast POST request */
+async requestCastSet(cast: Cast): Promise<HttpResponse<any>> {
+  if (environment.mockBackend) {
+    return this.mockBackend.requestCastSet(cast);
+  }
+  // Check if we have record of the cast and patch if we do
+  if (this.hasCast(cast.uuid)) {
+    // Do patch
+    let header = await this.headerUtil.generateHeader();
+    return this.http.delete(environment.backendURL + 'api/cast?castid=' + cast.uuid, {
+      headers: header,
+      observe: "response",
+      withCredentials: true
+    }).toPromise().catch((errorResp) => errorResp).then(
+        resp => this.respHandler.checkResponse<any>(resp)).then(async na => {
+      const rawCast = this.PatchPostPrep(cast);
+      return this.http.post(environment.backendURL + "api/cast", rawCast, {
         headers: header,
         observe: "response",
         withCredentials: true
-      }).toPromise().catch((errorResp) => errorResp).then((resp) => this.respHandler.checkResponse<any>(resp)).then(async na => {
-        let allSubCasts: RawSubCast[] = [];
-        let allPositions: Position[] = [];
-        Array.from(this.pieceAPI.pieces.values()).forEach(piece => {
-          allPositions.push(...piece.positions);
-        });
-        for (let filledPos of cast.filled_positions) {
-          for (let group of filledPos.groups) {
-            allSubCasts.push({
-              id: undefined,
-              positionId: Number(allPositions.find(val2 => {
-                return val2.uuid == filledPos.position_uuid;
-              }).uuid),
-              castNumber: group.group_index,
-              members: group.members.map(mem => {
-                return {
-                  id: undefined,
-                  userId: Number(mem.uuid),
-                  order: mem.position_number
-                }
-              })
-            });
-          }
-        }
-        let rawCast: RawCast = {
+      }).toPromise().catch((errorResp) => errorResp).then(
+          resp => this.respHandler.checkResponse<any>(resp));
+    });
+  } else {
+    // Do post
+    const rawCast = this.PatchPostPrep(cast);
+    let header = await this.headerUtil.generateHeader();
+    return this.http.post(environment.backendURL + "api/cast", rawCast, {
+      headers: header,
+      observe: "response",
+      withCredentials: true
+    }).toPromise().catch((errorResp) => errorResp).then(
+        resp => this.respHandler.checkResponse<any>(resp));
+  }
+}
+
+  private PatchPostPrep(cast: Cast): RawCast {
```

Comment:
Always use camelCase for function names rather than UpperCammelCase

Reply:
Fixed. Sorry.